mesonbuild/modules/unstable_xorg.py

```diff
@@ -0,0 +1,140 @@
+        'driver': '4',
+        'admin': '8',
+        'lib': '3',
+        'misc': '7',
+        'file': '5',
+    },
+    'sysv': {
+        'app': '1',
+        'driver': '7',
+        'admin': '1m',
+        'lib': '3',
+        'misc': '5',
+        'file': '4',
+    }
+}
+
+
+class XorgModule(NewExtensionModule):
+
+    """Module for upstream xorg development.
+
+    This module contains helper and abstractions fo upstream xorg projects to
+    use. These are *not* meant for projects using X11, though they might be
+    useful in that case.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.methods.update({
+            'format_man': self.format_man_method,
+        })
+
+    def _uses_sysv_man_sections(self) -> bool:
+        """Returns true if using sysv man sections."""
+        # XXX: This is only correct for build == host
+        # Copied from xorg-macros
+        return os.path.exists('/usr/share/man/man7/attributes.7')
+
+    @typed_pos_args('xorg.format_man', (str, mesonlib.File), str)
+    @typed_kwargs('xorg.format_man', KwargInfo('apploaddir', str, default=''))
+    def format_man_method(self, state: 'ModuleState', args: T.Tuple['mesonlib.FileOrString', str],
+                          kwargs: 'FormatManKW') -> ModuleReturnValue:
+        """Create a custom target that format's a man file.
+
+        This provides several nice features:
+        1. It knows about different man sections, which means that you don't
+           have to guess about sysv vs old unix layouts
+        2. it automatically sets some constant data (there's only one X server
+           at this point, and it's not going to bump to version 12 most likely)
+        3. it validates the sections
+
+        if the `apploaddir` keyword argument is unset then this will set any
+        __apploaddir__ values to an empty string.
+        """
+        sections = _MAN_SECTIONS['sysv' if self._uses_sysv_man_sections() else 'traditional']
+
+        rules: T.List[T.Tuple[str, str]] = [
+            ('__adminmansuffix__', sections['admin']),
+            ('__apploaddir__', kwargs['apploaddir']),
+            ('__appmansuffix__', sections['app']),
+            ('__drivermansuffix__', sections['driver']),
+            ('__filemansuffix__', sections['file']),
+            ('__libmansuffix__', sections['lib']),
+            ('__miscmansuffix__', sections['misc']),
+            ('__projectroot__', state.environment.get_prefix()),
+            ('__vendorversion__', f'"{state.project_version}" "X Version 11"'),
+            ('__xconfigfile__', 'xorg.conf'),
+            ('__xorgversion__', f'"{state.project_name} {state.project_version}" "X Version 11"'),
+            ('__xservername__', 'Xorg'),
+        ]
+
+        infile, section = args
+        if section not in sections:
+            raise InvalidArguments(
+                f'xorg-devel.format_man() second argument must be one of "{", ".join(sorted(sections))}", not "{section}"')
```

Comment:
This is unstable_xorg, what is "xorg-devel"?

Reply:
Ah, originally I called this `xorg-devel` because it's not for users of xorg, it's for actual xorg upstream. I dropped that because I didn't think it was useful, and I must have missed this usage of `xorg-devel`